xds/internal/balancer/ringhash/ringhash.go

```diff
@@ -417,6 +463,12 @@ func (cse *connectivityStateEvaluator) recordTransition(oldState, newState conne
 		updateVal := 2*uint64(idx) - 1 // -1 for oldState and +1 for new.
 		cse.nums[state] += updateVal
 	}
+	if oldState == connectivity.Shutdown {
```

Comment:
Can you ever transition _from_ shutdown?

Reply:
When a new SubConn is created, we record a shutdown->idle transition.
(And I believe we do the same in roundrobin)

Reply:
Ah yes, that makes sense thanks.